server/src/main/java/org/apache/kafka/server/AssignmentsManager.java

```diff
@@ -321,7 +321,7 @@ public void run() throws Exception {
                 AssignReplicasToDirsResponseData data = ((AssignReplicasToDirsResponse) response.responseBody()).data();
 
                 Set<AssignmentEvent> failed = filterFailures(data, inflight);
-                Set<AssignmentEvent> completed = Utils.diff(HashSet::new, inflight.values().stream().collect(Collectors.toSet()), failed);
+                Set<AssignmentEvent> completed = Utils.diff(HashSet::new, new HashSet<>(inflight.values()), failed);
```

Comment:
Maybe we don't need to create collection many times by skipping the failed elements.
```java
                Set<AssignmentEvent> failed = filterFailures(data, inflight);
                for (AssignmentEvent assignmentEvent : inflight.values()) {
                    if (failed.contains(assignmentEvent)) continue;
                    if (log.isDebugEnabled()) {
                        log.debug("Successfully propagated assignment {}", assignmentEvent);
                    }
                    assignmentEvent.onComplete();
                }
```

Reply:
I'd prefer only sticking to simple refactorings in this PR. We can do this optimization in another PR.